### PR TITLE
#129: Updated dependencies and fixed vulnerabilities

### LINKED ISSUE #129: Update dependencies and fix CVEs
### Situation

There are many outdated dependencies with possible CVEs. We should update those dependencies and fix vulnerabilities.

### Acceptance Criteria

- Dependencies are updated
- Vulnerabilities are fixed

### PR DESCRIPTION
**Pull request by morazow**
Fixes #129

### CODE REVIEW

File: doc/changes/changes_1.2.4.md [hunk 1]
```diff
@@ -0,0 +1,38 @@
+# The Spark Exasol Connector 1.2.4, released 2023-??-??
+
+Code name:
+
+## Summary
+
+## Refactoring
+
+* #129: Updated dependencies and fixed vulnerabilities
```
Comment (ckunki) [inline doc/changes/changes_1.2.4.md:9]:
```suggestion
* #129: Updated dependencies and fixed vulnerabilities

```

File: doc/changes/changes_1.2.4.md [hunk 2]
```diff
@@ -0,0 +1,38 @@
+# The Spark Exasol Connector 1.2.4, released 2023-??-??
+
+Code name:
+
+## Summary
```
Comment (ckunki) [inline doc/changes/changes_1.2.4.md:5]:
```suggestion
## Summary

Updated dependencies and fixed vulnerabilities.
```

File: doc/changes/changes_1.2.4.md [hunk 3]
```diff
@@ -0,0 +1,38 @@
+# The Spark Exasol Connector 1.2.4, released 2023-??-??
+
+Code name:
```
Comment (ckunki) [inline doc/changes/changes_1.2.4.md:3]:
```suggestion
Code name: Dependency Upgrade
```

File: doc/changes/changes_1.2.4.md [hunk 4]
```diff
@@ -0,0 +1,38 @@
+# The Spark Exasol Connector 1.2.4, released 2023-??-??
```
Comment (ckunki) [inline doc/changes/changes_1.2.4.md:1]: I assume updating dependencies only makes sense, when we release the spark connector -- please correct me if I'm wrong. 

Releasing would then require to define a release date. 
But in some cases I also created another branch eg. "release/1.2.4" and PR just for adjusting the release date in the changes file.
Maybe this is more convenient?

File: pom.xml [hunk 5]
```diff
@@ -16,15 +22,6 @@
...
     </distributionManagement>
-    <repositories>
-        <repository>
-            <id>maven.exasol.com</id>
-            <url>https://maven.exasol.com/artifactory/exasol-releases</url>
-            <snapshots>
-                <enabled>false</enabled>
-            </snapshots>
-        </repository>
-    </repositories>
```
Comment (ckunki) [inline pom.xml:27]: 👍

**Review by ckunki — CHANGES_REQUESTED** [review]

File: doc/changes/changes_1.2.4.md (hunk 3 above)
Comment (morazow) [reply doc/changes/changes_1.2.4.md:3]: Not defined yet, as mentioned above comment.

File: doc/changes/changes_1.2.4.md (hunk 4 above)
Comment (morazow) [reply doc/changes/changes_1.2.4.md:1]: I am not planning release yet,  I would like to investigate and fix also the #128, then release together with that.

Comment (ckunki) [reply doc/changes/changes_1.2.4.md:1]: 👍

**Review by ckunki — APPROVED** [review]

**Pull request merged**